fix: clarify recommended catch all modeling (#2471)
camunda/camunda-platform-docs: docs/components/modeler/bpmn/error-events/error-events.md

In BPMN, **errors** define possible errors that can occur. **Error events** are elements in the process referring to
defined errors. An error can be referenced by one or more error events.

An error must define an `errorCode`. The value of this `errorCode` is used to determine which catch event can catch the thrown error.

For throwing error events, it is possible to define the `errorCode` as an `expression`. When the event is reached,
the expression is evaluated. An error with the result of this expression is thrown. If no expression is used the
statically defined `errorCode` is used.

> For catching error events `errorCode` must be a static value.

Author: One thing I found confusing is that we mix the event (referencing the error) and the actual error. 

```suggestion
Errors that are meant to be caught must have a static `errorCode`.
```

On the other hand I understand that our users may not distinguish between error (can be caught or thrown) and the event (BPMN symbol) that references it.

> Alternatively a catching error event may omit the error reference all together. In this case it catches **all** thrown errors.

Reviewer: ❓ @nikku Why change the naming from error catch event to catching error event?

🔧 I suggest keeping `error catch event` for consistency, AFAIK we typically use the BPMN element type name

Author: I've tried to incorporate this into https://github.com/camunda/camunda-platform-docs/pull/2471/commits/0cea47e66891bcc1389575cefd12018a2472c087.

> Alternatively a catching error event may omit the error reference all together. In this case it catches **all** thrown errors.

Author: ```suggestion
Alternatively an error catch event may omit the error reference all together. In this case it catches **all** thrown errors.
```

It is not possible to define multiple error catch events with the same `errorCode` in a single scope. It is also not
permitted to have multiple error catch events without an `errorCode` in a single scope. The deployment gets rejected in
these cases. However, it is possible to define both an error catch event referencing an error with a particular `errorCode` and an error catch-all event within the same scope. When this happens, the error catch event

> that matches the `errorCode` is prioritized.

Reviewer: 👍 I like the changes, but I think there's another sentence that needs to be adjusted

>It is also not permitted to have multiple error catch events without an `errorCode` in a single scope.

🔧 Is that sentence still relevant? I believe it should say something like:

```
It is also not permitted to have multiple error catch-all events in a single scope.
```

Author: I've tried to incorporate this into https://github.com/camunda/camunda-platform-docs/pull/2471/commits/0cea47e66891bcc1389575cefd12018a2472c087.

<bpmn:boundaryEvent id="invalid-credit-card-1" name="Invalid Credit Card" attachedToRef="collect-money">
 <bpmn:errorEventDefinition errorRef="invalid-credit-card-error" />
</bpmn:boundaryEvent>
```

> A catch-all error boundary event without:

Author: ```suggestion
A error boundary catch-all event:
```
